src/kixi/datastore/filestore/command_handler.clj

```diff
@@ -10,21 +11,54 @@
 (sh/alias 'up 'kixi.datastore.filestore.upload)
 (sh/alias 'up-reject 'kixi.event.file.upload.rejection)
 
+(def small-file-size 10000000) ;; 10MB
+(def largest-file-size 5000000000) ;; 5GB
+
 (defn uuid
   []
   (str (java.util.UUID/randomUUID)))
 
+(s/def ::start-byte
+  (s/with-gen int?
+    #(gen/such-that (partial <= 0) (gen/large-integer) 100)))
+
+(s/def ::length-bytes
+  (s/with-gen int?
+    #(gen/such-that (fn [x] (<= 0 x small-file-size)) (gen/large-integer) 100)))
+
+(s/def ::file-size (s/int-in 0 largest-file-size))
+
+(s/fdef calc-chunk-ranges
+        :args (s/cat :file-size ::file-size)
+        :fn (fn [{{:keys [file-size]} :args ret :ret}]
```

Comment:
Want to try the new 'all-preds' way of writing these from kixi.spec?

Reply:
Is it quick?